### PR TITLE
bug fix

### LINKED ISSUE #71: 他人の読んだよマークが消せる
消せちゃあかん

### PR DESCRIPTION
**Pull request by suzan2go**
* fix #71 #70 

先にこっちマージしてもらえるといい感じになります #73

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -7,9 +7,7 @@ class ApplicationController < ActionController::Base
   private
 
   def login_required
-    if session[:user_id]
-      @current_user = User.find(session[:user_id])
-    end
+    redirect_to root_path, notice: "ログインしてください" unless current_user
```
Comment (nasum) [inline app/controllers/application_controller.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: spec/factories/poems.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :poem do
+    title "hogehoge"
```
Comment (nasum) [inline spec/factories/poems.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: spec/factories/poems.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+    description "fugafuga"
```
Comment (nasum) [inline spec/factories/poems.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: spec/controllers/poems_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe PoemsController, type: :controller do
+  describe 'GET #show' do
+    let(:poem) { create(:poem) }
+    subject { get :show , id: poem.id }
```
Comment (nasum) [inline spec/controllers/poems_controller_spec.rb]: __[rubocop]__
* Space found before comma.

File: spec/controllers/poems_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    context 'before login' do
```
Comment (nasum) [inline spec/controllers/poems_controller_spec.rb]: __[rubocop]__
* Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

File: spec/controllers/poems_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    context 'before login' do 
+      it { is_expected.to redirect_to(root_path) }
+    end
+  end
+
+  describe 'GET #show', login_as: :current_user do
+    let(:poem) { create(:poem) }
+    let(:current_user) { create(:user) }
+    subject { get :show , id: poem.id }
```
Comment (nasum) [inline spec/controllers/poems_controller_spec.rb]: __[rubocop]__
* Space found before comma.

File: spec/controllers/poems_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+    context 'afer login' do
```
Comment (nasum) [inline spec/controllers/poems_controller_spec.rb]: __[rubocop]__
* Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

File: config/routes.rb [hunk 8]
```diff
@@ -5,8 +5,9 @@
   get  '/logout' => 'sessions#destroy'
 
   resources :home
-  resources :poems
-  resources :read_poems, only: [:create, :destroy]
+  resources :poems, shallow: true do
```
Comment (nasum) [inline config/routes.rb]: __[rails_best_practices]__
* restrict auto-generated routes poems (except: [:destroy]) [:ref](http://rails-bestpractices.com/posts/2011/08/19/restrict-auto-generated-routes/)

File: spec/controllers/poems_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe PoemsController, type: :controller do
+  describe 'GET #show' do
+    let(:poem) { create(:poem) }
+    subject { get :show, id: poem.id }
+
+    context 'before login' do
```
Comment (nasum) [inline spec/controllers/poems_controller_spec.rb]: __[rubocop]__
* Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

**Comment by nasum** [comment]
なんかテスト落ちてますな

**Comment by suzan2go** [comment]
@nasum rebased !

**Comment by nasum** [comment]
@suzan2go Yeah!! LGTM !!

**Pull request merged**